cmd/mt-gateway/api.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+//Maintains a set of `http.Handlers` for the different API endpoints.
+//Used to generate an http.ServeMux via `api.Mux()`
+type Api struct {
+	ingestHandler     http.Handler
+	metrictankHandler http.Handler
+	graphiteHandler   http.Handler
+	bulkImportHandler http.Handler
+}
+
+//Constructs a new Api based on the passed in URLS
+func NewApi(urls Urls) Api {
+	api := Api{}
+	//TODO implement actual kafka based import handler
+	api.ingestHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotImplemented)
+		_, _ = fmt.Fprintln(w, "http ingest not yet implemented")
+	})
+	api.graphiteHandler = newProxyWithLogging("graphite", urls.graphite)
+	api.metrictankHandler = newProxyWithLogging("metrictank", urls.metrictank)
+	api.bulkImportHandler = newProxyWithLogging("bulk-importer", urls.bulkImporter)
+	return api
+}
+
+//Builds an http.ServeMux based on the handlers defined in the Api
+func (api Api) Mux() *http.ServeMux {
+	mux := http.NewServeMux()
+	//By default everything is proxied to graphite
+	//This includes endpoints under `/metrics` which aren't explicitly rerouted
+	mux.Handle("/", api.graphiteHandler)
+	//`/metrics` is handled locally by the kafka ingester (not yet implemented)
+	mux.Handle("/metrics", api.ingestHandler)
+	//other endpoints are proxied to metrictank or mt-whisper-import-writer
+	mux.Handle("/metrics/index.json", api.metrictankHandler)
+	mux.Handle("/metrics/delete", api.metrictankHandler)
+	mux.Handle("/metrics/import", api.bulkImportHandler)
```

Comment:
in tsdb we only register this url when there is actually an importer url defined, because this one is optional. should we maybe do that here too?

Reply:
How would you feel about changing up the implementation a bit?

I'm thinking we always add the endpoint, but if the url isn't defined return a 503 + message explaining that no importer is defined

Reply:
yeah, that's fine too, actually that's better because it can return an explanatory message

Reply:
5b18626628214310b7b15baa61bf1d5dc61119e5